ee/server/service/hosts.go

```diff
@@ -235,14 +235,22 @@ func (svc *Service) enqueueLockHostRequest(ctx context.Context, host *fleet.Host
 }
 
 func (svc *Service) enqueueUnlockHostRequest(ctx context.Context, host *fleet.Host, lockStatus *fleet.HostLockWipeStatus) (string, error) {
+	if lockStatus.HostFleetPlatform == "darwin" && !lockStatus.UnlockRequestedAt.IsZero() {
+		// unlock request was already recorded, just return the unlock PIN as this
+		// is just a request to view it again.
+		return lockStatus.UnlockPIN, nil
```

Comment:
Shouldn't we still create an activity item every time the PIN is viewed? That was my understanding
but I may have missed something!

Reply:
great catch, yes we should!

Reply:
Argh yeah I think you're right, I'll change that first thing tomorrow.